tests/test_project.py

```diff
@@ -368,6 +398,74 @@ def test_find_jobs_logical_operators(self):
         q = {'$or': [{'$and': [{'b': {'c': 0}}, {'b': {'c': 1}}]}]}
         assert len(self.project.find_jobs(q)) == 0
 
+        # explicit sp.-prefix
+        assert len(self.project.find_jobs({'$and': [{}, {'sp.a': 0}]})) == 1
+        assert len(self.project.find_jobs({'$or': [{}, {'sp.a': 0}]})) == len(self.project)
+        q = {'$and': [{'sp.a': 0}, {'sp.a': 1}]}
+        assert len(self.project.find_jobs(q)) == 0
+        q = {'$or': [{'sp.a': 0}, {'sp.a': 1}]}
+        assert len(self.project.find_jobs(q)) == 2
+        q = {'$and': [{'$and': [{'sp.a': 0}, {'sp.a': 1}]}]}
+        assert len(self.project.find_jobs(q)) == 0
+        q = {'$and': [{'$or': [{'sp.a': 0}, {'sp.a': 1}]}]}
+        assert len(self.project.find_jobs(q)) == 2
+        q = {'$or': [{'$or': [{'sp.a': 0}, {'sp.a': 1}]}]}
+        assert len(self.project.find_jobs(q)) == 2
+        q = {'$or': [{'$and': [{'sp.a': 0}, {'sp.a': 1}]}]}
+        assert len(self.project.find_jobs(q)) == 0
+        assert len(self.project.find_jobs({'$and': [{}, {'sp.b': {'c': 0}}]})) == 1
+        assert len(self.project.find_jobs({'$and': [{}, {'sp.b.c': 0}]})) == 1
+        assert len(self.project.find_jobs(
+            {'$or': [{}, {'sp.b': {'c': 0}}]})) == len(self.project)
+        assert len(self.project.find_jobs(
+            {'$or': [{}, {'sp.b.c': 0}]})) == len(self.project)
+        q = {'$and': [{'sp.b': {'c': 0}}, {'sp.b': {'c': 1}}]}
+        assert len(self.project.find_jobs(q)) == 0
+        q = {'$and': [{'sp.b': {'c': 0}}, {'sp.b.c': 1}]}
+        assert len(self.project.find_jobs(q)) == 0
+        q = {'$or': [{'sp.b': {'c': 0}}, {'sp.b': {'c': 1}}]}
+        assert len(self.project.find_jobs(q)) == 2
+        q = {'$or': [{'sp.b': {'c': 0}}, {'sp.b.c': 1}]}
+        assert len(self.project.find_jobs(q)) == 2
+        q = {'$and': [{'$and': [{'sp.b': {'c': 0}}, {'sp.b': {'c': 1}}]}]}
+        assert len(self.project.find_jobs(q)) == 0
+        q = {'$and': [{'$and': [{'sp.b.c': 0}, {'sp.b.c': 1}]}]}
+        assert len(self.project.find_jobs(q)) == 0
+        q = {'$and': [{'$or': [{'sp.b': {'c': 0}}, {'sp.b': {'c': 1}}]}]}
+        assert len(self.project.find_jobs(q)) == 2
+        q = {'$and': [{'$or': [{'sp.b.c': 0}, {'sp.b.c': 1}]}]}
+        assert len(self.project.find_jobs(q)) == 2
+        q = {'$or': [{'$or': [{'sp.b': {'c': 0}}, {'sp.b': {'c': 1}}]}]}
+        assert len(self.project.find_jobs(q)) == 2
+        q = {'$or': [{'$or': [{'sp.b.c': 0}, {'sp.b.c': 1}]}]}
+        assert len(self.project.find_jobs(q)) == 2
+        q = {'$or': [{'$and': [{'sp.b': {'c': 0}}, {'sp.b': {'c': 1}}]}]}
+        assert len(self.project.find_jobs(q)) == 0
+        q = {'$or': [{'$and': [{'sp.b.c': 0}, {'sp.b.c': 1}]}]}
+        assert len(self.project.find_jobs(q)) == 0
+
+        # Mixed filters
+        def assert_result_len(q, num):
+            assert len(self.project.find_jobs(q)) == num
```

Comment:
I like this. Maybe use that also for the tests above?

Reply:
Sure 👍

Reply:
Done 👍